cart/domain/cart/discount.go

```diff
@@ -111,6 +116,25 @@ func (i *Item) HasAppliedDiscounts() (bool, error) {
 	return len(discounts) > 0, nil
 }
 
+// MergeDiscounts parses discounts of a shipping shipping
+// All discounts with the same type and title are aggregated and returned as one with a summed price
+func (s *ShippingItem) MergeDiscounts() (AppliedDiscounts, error) {
+	sort.SliceStable(s.AppliedDiscounts, func(x, y int) bool {
+		return s.AppliedDiscounts[x].SortOrder < s.AppliedDiscounts[y].SortOrder
+	})
+
+	return s.AppliedDiscounts, nil
+}
+
```

Comment:
You could try to use a more abstract private function for MergeDiscounts() for Items and Shipping Items
as they basically have the same code now